### DISCUSSION TITLE
how to automatically set the latest copied item as active

### DISCUSSION THREAD

**Discussion by litoj** [hluk/CopyQ#2526] [General]
I use sway but everythings works just fine. But I couldn't find my way to make copyq automatically provide the last copied item. I always had to open the app and click the `Activate items` (`Aktivovat prvky` in `cs` locale) button. But I would prefer not having to open the application, and since it does pick up the copied text immediately I assume it must be also capable of providing that data without any unnecessary extra steps.

I saw with clipman people had to use something like `wl-paste watch -p clipman store -P`. 

Also, how can I set the filepath for storing the clipboard? I like using `/tmp` to keep the data only until restart.

**Comment by litoj** [comment]
Also, I just noticed, is there a way to not open the window after *starting the server* as it calls it?

**Reply by hluk** [reply]
The default installed app desktop file (usually `/usr/share/applications/com.github.hluk.copyq.desktop`) uses `copyq --start-server show` command to start the app - this shows the app window after start. I think it should be possible to copy the desktop file to `~/.local/share/applications/` and change the command in the file (`Exec=copyq`) so that the main window stays hidden after start.

**Comment by hluk** [comment]
> But I couldn't find my way to make copyq automatically provide the last copied item.

Not sure if I understand. If you want to copy the last item stored in CopyQ back to the system clipboard when the app starts, you can use a command to start the app:

```bash
copyq --start-server 'copy(read(0))'
```

Or create a command so the app does that automatically at start ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Command="
    var onStart_ = global.onStart;
    global.onStart = function() {
        onStart_();
        copy(read(0));
    }"
Icon=\xf46d
IsScript=true
Name=Copy the last item at start
```

> I saw with clipman people had to use something like `wl-paste watch -p clipman store -P`.

The `wl-paste --watch ...` command is to run a command on selection/clipboard change. CopyQ should be able to monitor the changes itself (at least in Sway and some other Wayland compositors).

> Also, how can I set the filepath for storing the clipboard? I like using `/tmp` to keep the data only until restart.

This is possible with [Synchronize plugin](https://copyq.readthedocs.io/en/latest/synchronize.html) or you can disable "Save Items" option for a tab (in Preferences under "Tabs" section) store tab data only in the main memory and not on disk.

**Reply by litoj** [reply]
By copying the last/first item I wanted to provide the latest item in clipboard automatically. An example might help clarify my intentions:

1. copy text in an app
2. close the app
3. clipboard acts as empty (expected to paste the copied text)
4. workaround: open copyq and doubleclick the first item

I suppose the technical goal is to always keep `copyq` as the clipboard provider - if an app offers some data, immediately offer that data back.

I feel like this is a common usecase, so I am probably missing some option made specifically for this, I was just unable to find it despite my best efforts.

**Reply by litoj** [reply]
> disable "Save Items" option for a tab

This worked, thank you. Although I got very confused by the difference in behaviour with `General options > Clipboard manipulation > save clipboard`.

**Reply by egor230** [reply]
copyq:

select(0)

**Reply by litoj** [reply]
> `select(0)`

This looks like what I wanted, but where do I put this to get it executed on every clipboard change?

**Reply by egor230** [reply]
> > `select(0)`
> 
> Это похоже на то, что я хотел, но куда мне поместить это, чтобы оно выполнялось при каждом изменении буфера обмена?


  if [ -z "$(copyq read 0)" ]; then
   echo "empty"
   copyq select 0
   sleep 6
  else
   echo "full"
   sleep 2
  fi

**Reply by litoj** [reply]
So there's no actual way to create a hook on copyq registering a new item?
